src/lager_util.erl

```diff
@@ -460,19 +460,21 @@ i3l(I)              -> integer_to_list(I).
 
 %% When log_root option is provided, get the real path to a file
 expand_path(RelPath) ->
-    case application:get_env(lager, log_root) of
-        {ok, LogRoot} when is_list(LogRoot) -> % Join relative path
-            %% check if the given RelPath contains LogRoot, if so, do not add
-            %% it again; see gh #304
-            case string:str(filename:dirname(RelPath), LogRoot) of
-                X when X > 0 ->
-                    RelPath;
-                _Zero ->
-                    filename:join(LogRoot, RelPath)
-            end;
-        undefined -> % No log_root given, keep relative path
-            RelPath
-    end.
+    RelPath2 = case application:get_env(lager, log_root) of
+                   {ok, LogRoot} when is_list(LogRoot) -> % Join relative path
+                       %% check if the given RelPath contains LogRoot, if so, do not add
+                       %% it again; see gh #304
+                       case string:str(filename:dirname(RelPath), LogRoot) of
+                           X when X > 0 ->
```

Comment:
This seems to work badly for e.g. `console.log`  and `log`, since root is part of the former, in terms of matching.

Reply:
Yes, `string:str` is not so precise, this change commited in #317 to fix #304.
this pr change nothing with it.  

------

maybe should make a new pr or issue to specially discuss this,
 i saw https://github.com/erlang-lager/lager/issues/304#issuecomment-164807302 said should trans path to abspath. not not do this finally, maybe there has some other to avoid

Reply:
Yeah, sorry, my bad. I didn't mean to imply your pull request changed it in any way. It was simply the most recent one I found "dealing" with this.

I think ```https://github.com/erlang-lager/lager/commit/b9ce226bff4b47f93a4aeac6bbd34bcf78412a8d``` might to the trick: it's part of another pull request I'm currently preparing. Maybe @mrallen1 can help out, here.

Reply:
The origin might disappear so I'm pasting the code here `filename:dirname(RelPath) =:= filename:basename(LogRoot)`: it basically compares the first folder.